functions/src/index.ts

```diff
@@ -15,6 +15,39 @@ const thingspeakUrl = (channelId: string) =>
   `https://api.thingspeak.com/channels/${channelId}/feeds.json`;
 const readingsSubcollection = (docId: string) => `/sensors/${docId}/readings`;
 
+interface pm25BufferElement {
+  timestamp: FirebaseFirestore.Timestamp | null;
+  channelAPm25: number;
+  channelBPm25: number;
+  humidity: number;
+  latitude: number;
+  longitude: number;
+}
+
+const defaultPm25BufferElement: pm25BufferElement = {
+  timestamp: null,
+  channelAPm25: NaN,
+  channelBPm25: NaN,
+  humidity: NaN,
+  latitude: NaN,
+  longitude: NaN,
+};
+
+interface AQIBufferElement {
+  timestamp: FirebaseFirestore.FieldValue | null;
+  aqi: number;
+}
+const defaultAQIBufferElement: AQIBufferElement = {
+  timestamp: null,
+  aqi: NaN,
+};
+
+enum bufferStatus {
+  Exists = 1 /* eslint-disable-line no-magic-numbers */,
```

Comment:
Can `bufferStatus` be zero indexed and then we don't need a magic number disabling? The [TypeScript handbook](https://www.typescriptlang.org/docs/handbook/enums.html ) seems like it will auto-start at 0 if we leave out an assignment, but I'm not sure if there is a Firebase reason to start at 1. Also, can this be a `const  enum`, not that I fully understand the difference?

Reply:
Yep, good suggestion.